test/common/secret/sds_api_test.cc

```diff
@@ -0,0 +1,161 @@
+#include <memory>
+
+#include "envoy/api/v2/auth/cert.pb.h"
+#include "envoy/common/exception.h"
+
+#include "common/secret/sds_api.h"
+
+#include "test/mocks/grpc/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using ::testing::Invoke;
+using ::testing::Return;
+using ::testing::_;
+
+namespace Envoy {
+namespace Secret {
+namespace {
+
+class MockServer : public Server::MockInstance {
+public:
+  Init::Manager& initManager() { return initmanager_; }
+
+private:
+  class InitManager : public Init::Manager {
+  public:
+    void initialize(std::function<void()> callback);
+    void registerTarget(Init::Target&) override {}
+  };
+
+  InitManager initmanager_;
+};
+
+class SdsApiTest : public testing::Test {
+public:
+  Grpc::MockAsyncClient* grpc_client_{new Grpc::MockAsyncClient};
+  Grpc::MockAsyncClientFactory* factory_{new Grpc::MockAsyncClientFactory};
+};
+
+TEST_F(SdsApiTest, BasicTest) {
+  ::testing::InSequence s;
+  Server::MockInstance server;
```

Comment:
Wrap it with NiceMock<> to eliminate these warnings:

Uninteresting mock function call - taking default action specified at:
test/mocks/server/mocks.cc:124:
    Function call: random()
          Returns: @0x7ffff7fc3ff8 8-byte object <F8-F5 C6-04 00-00 00-00>
unknown file: Failure